app/src/main/java/org/schabi/newpipe/player/VideoPlayerImpl.java

```diff
@@ -1508,7 +1517,84 @@ public void setControlsSize() {
                 ? Gravity.START : Gravity.END)
                 : Gravity.TOP;
 
-        getTopControlsRoot().getLayoutParams().width = width;
+        final int navBarHeight = DeviceUtils.hasNavBar(windowManager) ? getNavBarHeight() : 0;
+        final int statusBarHeight = getStatusBarHeight();
+        final int controlsPadding = service.getResources()
+                .getDimensionPixelSize(R.dimen.player_main_controls_padding);
+        final int playerTopPadding = service.getResources()
+                .getDimensionPixelSize(R.dimen.player_main_top_padding);
+
+        // Handle rotation
+        final int ctrlPadLeft;
+        final int ctrlPadRight;
+        final int ctrlPadTop = playerTopPadding + statusBarHeight;
+        final int ctrlPadBottom = navBarHeight + insetBottom;
+        final int bottom = ctrlPadBottom + (navBarHeight == 0 ? controlsPadding : playerTopPadding);
+        final int rotation = windowManager.getDefaultDisplay().getRotation();
+
+        if (DeviceUtils.isTv(service)) {
+            gotCutout = false;
+            ctrlPadLeft = controlsPadding;
+            ctrlPadRight = controlsPadding;
+        } else if (DeviceUtils.isTablet(service)) {
+            if (gotCutout) {
+                ctrlPadLeft = controlsPadding + insetLeft;
+                ctrlPadRight = controlsPadding + insetRight;
+            } else {
+                ctrlPadLeft = controlsPadding;
+                ctrlPadRight = controlsPadding;
+            }
+        } else {
+            if (rotation == Surface.ROTATION_90) {
+                if (gotCutout) {
+                    ctrlPadLeft = controlsPadding + insetLeft;
+                    ctrlPadRight = controlsPadding
+                            + (getNavBarMode() == 2 ? 0 : navBarHeight) + insetRight;
+                } else {
+                    ctrlPadLeft = controlsPadding;
+                    ctrlPadRight = controlsPadding;
+                }
+            } else {
+                if (gotCutout) {
+                    ctrlPadLeft = controlsPadding
+                            + (getNavBarMode() == 2 ? 0 : navBarHeight) + insetLeft;
+                    ctrlPadRight = controlsPadding + insetRight;
+                } else {
+                    ctrlPadLeft = controlsPadding;
+                    ctrlPadRight = controlsPadding;
+                }
+            }
```

Comment:
why do you only check for 90 deg, but not for 180?

Reply:
@TobiGr When it's 90° in landscape then the navbar is on the right side of your screen, so the else statement handles the opposite. And this orientation check only applies to phones in landscape, so it always seems to work.

If you look at #4272 then you'll see that avently worked on a cleaner method, but we had a disagreement about handling a navbar feature that's found on a lot of custom Android 10 ROMs (LineageOS, ArrowOS, Resurrection Remix, PixelExperience, AICP etc).